### PR TITLE
finishing touches on last 9.x postgres

### PR DESCRIPTION
**Pull request by teodorescuserban**
- `pg_example` is reserved... :)
- replaced `'*'` with `'0.0.0.0'` to get rid of a complaint message saying it cannot bind to the IPv6 address.
- batteries included this time. we will probably want to drop a lot of env variables and configs mapped from the stacks to the container.

### CODE REVIEW

File: postgis/run_postgres [hunk 1]
```diff
@@ -1,42 +1,33 @@
...
-# add a user, pass and a database owned by that user
-TMP_FILE=/tmp/pgsql.initialize.txt
-# also add postgis extensions to that db
-TMP_GIS_FILE=/tmp/pggis.initialize.txt
-if [ ! -f $TMP_FILE ]; then
-    # create user and db
-    PGSQL_USER=${PGSQL_USER:-"dbuser"}
-    PGSQL_PASS=${PGSQL_PASS:-"dbpass"}
-    PGSQL_DB=${PGSQL_DB:-"dbname"}
-    cat > $TMP_FILE << EOSQL
```
Comment (cafuego) [inline postgis/run_postgres:24]: test -n ${PGDATA} || exit 1

File: postgis/run_postgres [hunk 2]
```diff
@@ -1,42 +1,33 @@
...
+mkdir -p $PGDATA
+
+# if data folder is empty
+if [ "$(find $PGDATA ! -path $PGDATA | wc -l)" == "0" ]; then
```
Comment (cafuego) [inline postgis/run_postgres]: I suspect this is very slow if there are a lot of files under $PGDATA; find will try to list all of them before piping the output to `wc`. You should add a `-maxdepth` param or be able to use `ls -A $PGDATA | wc -l` instead. That will *not* list dotfiles, but if there is data, there should be a subdir that is not a dotfile.

File: postgis/run_postgres [hunk 3]
```diff
@@ -1,42 +1,33 @@
...
-    s6-setuidgid postgres pg_ctl -w start
-    s6-setuidgid postgres psql -a -f $TMP_FILE
-    s6-setuidgid postgres psql -a -U $PGSQL_USER $PGSQL_DB -f $TMP_GIS_FILE
-    s6-setuidgid postgres pg_ctl stop
-fi
+EOF
 
-args="-D ${PGDATA}"
-[ -z ${PG_CONFIG_FILE} ] || args="--config-file=${PG_CONFIG_FILE}"
+    sleep 2
```
Comment (cafuego) [inline postgis/run_postgres:28]: Is that enough?

File: postgis/run_postgres [hunk 4]
```diff
@@ -1,42 +1,33 @@
...
+    s6-setuidgid postgres pg_ctl -w stop
+
+fi
 
-exec s6-setuidgid postgres /usr/bin/postgres -h 0.0.0.0 $args
+exec s6-setuidgid postgres /usr/bin/postgres -h 0.0.0.0 -D ${PGDATA} --config-file=${PG_CONFIG_FILE}
```
Comment (cafuego) [inline postgis/run_postgres:33]: I'll believe you :-)

**Review by cafuego — CHANGES_REQUESTED** [review]
Only one change, to make startup be fast with a big datadir.

File: postgis/run_postgres (hunk 2 above)
Comment (teodorescuserban) [reply postgis/run_postgres]: `ls -A` sucked for some reason. I'll add `-maxdepth`

File: postgis/run_postgres (hunk 3 above)
Comment (teodorescuserban) [reply postgis/run_postgres:28]: added `-w` flag that would make it wait until the operation is finished, so `sleep` is no longer needed.

File: postgis/run_postgres (hunk 4 above)
Comment (teodorescuserban) [reply postgis/run_postgres:33]: proof:
https://github.com/UN-OCHA/docker-images/blob/998985cb4635e3deb8c0ebf83a55a08fd6eafff6/postgis/Dockerfile.tmpl#L31
and
https://github.com/UN-OCHA/docker-images/blob/998985cb4635e3deb8c0ebf83a55a08fd6eafff6/postgis/Dockerfile.tmpl#L32

File: postgis/run_postgres (hunk 1 above)
Comment (teodorescuserban) [reply postgis/run_postgres:24]: not used anymore.

**Comment by teodorescuserban** [comment]
@cafuego, fixed. merge (or complain) at your own leisure ;)

**Pull request merged**